shared/src/transport/http.rs

```diff
@@ -24,26 +24,29 @@ struct Inner {
     url: Url,
     id: AtomicUsize,
     metrics: &'static TransportMetrics,
+    /// Name of the transport used in logs to distinguish different transports.
+    name: String,
 }
 
 impl HttpTransport {
-    pub fn new(client: Client, url: Url) -> Self {
+    pub fn new(client: Client, url: Url, name: String) -> Self {
         Self {
             client,
             inner: Arc::new(Inner {
                 url,
                 id: AtomicUsize::new(0),
                 metrics: TransportMetrics::instance(get_metric_storage_registry()).unwrap(),
+                name,
```

Comment:
Maybe for now, we can default name to the URL to avoid having more things to configure? Not sure if it makes the logs too noisy though.

Reply:
The current default is in the index in the argument so that there is nothing to configure. I think this is fine. Mostly wanted to explain why I rejected the other way. If we see in the future that we would really like custom names we can still make it configurable.